src/Compilers/Core/AnalyzerDriver/DiagnosticDescriptorExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class DiagnosticDescriptorExtensions
+    {
+        public static void SetOnLocalizableStringException(this DiagnosticDescriptor descriptor, DiagnosticAnalyzer owner, Action<Exception, DiagnosticAnalyzer, Diagnostic> onLocalizableStringException)
```

Comment:
since descriptor is created once and cached for the life time of analyzer, I think we don't need to worry about allocation that much.